src/Model/Profile.php

```diff
@@ -643,27 +643,29 @@ public static function getEventsReminderHTML()
 		$bd_format = L10n::t('g A l F d'); // 8 AM Friday January 18
 		$classtoday = '';
 
+		$self = dba::selectFirst('contact', ['id'], ['uid' => local_user(), 'self' => true]);
+
 		$s = dba::p(
 			"SELECT *
 			FROM `event`
-			WHERE `event`.`uid` = ?
-			AND  `event`.`type` != 'birthday'
-			AND  `event`.`start` < ?
-			AND  `event`.`start` >= ?
-			AND NOT EXISTS (
-				SELECT `id`
-				FROM `item`
-				WHERE `item`.`uid` = `event`.`uid`
+			JOIN `item`
+				ON `item`.`uid` = `event`.`uid`
 				AND `item`.`parent-uri` = `event`.`uri`
-				AND `item`.`verb` = ?
-				AND `item`.`visible`
-				AND NOT `item`.`deleted`
-			)
+			WHERE `event`.`uid` = ?
+			AND `event`.`type` != 'birthday'
+			AND `event`.`start` < ?
+			AND `event`.`start` >= ?
+			AND `item`.`author-id` = ?
```

Comment:
The ```author-id``` (and ```owner-id```) are the contact id of the representing contact with the ```uid=0```.

Reply:
I'm not sure what you're asking me to change here. The resulting attendance activity item has `uid=1` and `author-id=6`, which are respectively my user id and my self-contact id.

Reply:
If the "contact" record with "id=6" really has got the "uid=1" and "self=true" then your system is in trouble.

See here: https://github.com/friendica/friendica/blob/develop/src/Model/Item.php#L481

The "author-id" should always be a contact record with "uid=0".

Reply:
Aaaah, I remember now the issue with matching a `local_user()` id with the associated public contact. I probably did wrong then.

Yep, it doesn't work.

Groumpf.

Reply:
I KNEW SOMETHING EXISTED!

I wrote it: `public_contact()`.